pandas/tests/types/test_validate.py

```diff
@@ -0,0 +1,169 @@
+from pandas.types.validate import _validate_bool_type
+from unittest import TestCase
```

Comment:
@jreback how should we handle tests unrelated to the series and the dataframe, for instance the `Categorical` class? Should we add an additional test_validate file in the tests/ directory to account for those that don't fit under series and frame? Similarly I'd assume we could place put the `eval` tests inside computation/tests/test_validate or something like that. Thanks!